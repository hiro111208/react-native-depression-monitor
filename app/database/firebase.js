//import firebase from "firebase/app";
import "firebase/firestore";
import firebase from 'firebase'
require('firebase/auth')

const firebaseConfig = {
  apiKey: "AIzaSyDwOUkdxC7kNNILoFTSx05PTUxRci_kcaw",
  authDomain: "fireship-tutorial-646fc.firebaseapp.com",
  projectId: "fireship-tutorial-646fc",
  storageBucket: "fireship-tutorial-646fc.appspot.com",
  messagingSenderId: "289064122528",
  appId: "1:289064122528:web:ee39abfa2183c8499071c5",
  measurementId: "G-D35RDH9V0E",
};

firebase.initializeApp(firebaseConfig);

export default firebase;
