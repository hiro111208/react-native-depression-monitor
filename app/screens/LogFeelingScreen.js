import React, { useState, useEffect } from "react";
import {
  StyleSheet,
  Text,
  View,
  TouchableOpacity,
  Alert,
  ScrollView,
} from "react-native";
import firebase from "../database/firebase";
import FeelingsRadioButtons from "../src/components/FeelingsRadioButtons";
import FeelingsSlider from "../src/components/FeelingsSlider";
import colors from "../config/colors";
import * as indexStyles from "../config/indexStyles";

/**
 * Screen where the user logs their feelings. Users will
 * log their feelings before and after their therapy
 */
function LogFeelingScreen({ navigation, route }) {
  const [overallFeeling, setOverallFeeling] = useState("");
  const [paranoid, setParanoid] = useState(1);
  const [anxious, setAnxious] = useState(1);
  const [sad, setSad] = useState(1);
  const [friendly, setFriendly] = useState(1);
  const [user, setUser] = useState(undefined);

  //Get userID corresponding to current user
  function getUser() {
    firebase
      .firestore()
      .collection("users")
      .doc(firebase.auth().currentUser.uid)
      .get()
      .then((doc) => {
        setUser(doc.data());
      })
      .catch((error) => {
        console.log("Error getting document:", error);
      });
  }

  useEffect(() => {
    getUser();
  }, []);

  //Reset all states
  const reset = () => {
    setAnxious(1);
    setParanoid(1);
    setFriendly(1);
    setOverallFeeling("");
    setSad(1);
    setUser(undefined);
  };

  //Store the user's feelings in the database
  function saveFeelings() {
    getUser();
    firebase
      .firestore()
      .collection("feelings")
      .doc()
      .set({
        overall: overallFeeling,
        paranoid: paranoid,
        friendly: friendly,
        sad: sad,
        anxious: anxious,
        userID: user.userID,
        timeStamp: new Date(),
      })
      .then(() => {
        console.log("Feelings saved");
      })
      .catch((error) => {
        console.error("Error saving feelings: ", error);
      });
  }

  //Handle when continue button is pressed
  //Go to either therapy screen or patient dashboard depending on whether therapy has started or ended
  function handleOnContinuePress() {
    if (overallFeeling === "") {
      Alert.alert("Please fill in how you are feeling!");
    } else {
      if (route.params.cameFrom === "HomeScreen") {
        saveFeelings();
        reset();
        navigation.navigate("TherapyScreen", {
          onGoBack: () => route.params.onGoBack(),
        });
      } else {
        saveFeelings();
        reset();
        navigation.navigate("PatientDashboard");
      }
    }
  }

  //Render the log feelings screen
  return (
    <View style={styles.container}>
      {/* Render header  */}
      <View style={[styles.header, indexStyles.centering]}>
        <Text style={styles.headerText}>How are you feeling?</Text>
      </View>

      {/* Render feeling emojis as radio buttons */}
      <FeelingsRadioButtons
        setOverallFeeling={setOverallFeeling}
        overallFeeling={overallFeeling}
      />

      {/* Render sliders for each emotion */}
      <ScrollView style={styles.slidersContainer}>
        <View style={[styles.topAndBottom, indexStyles.centering]}>
          <Text style={indexStyles.textGrey}>Paranoid</Text>
          <FeelingsSlider
            setFeelingState={setParanoid}
            feelingState={paranoid}
          />
        </View>

        <View style={[styles.topAndBottom, indexStyles.centering]}>
          <Text style={indexStyles.textGrey}>Anxious</Text>
          <FeelingsSlider setFeelingState={setAnxious} feelingState={anxious} />
        </View>

        <View style={[styles.topAndBottom, indexStyles.centering]}>
          <Text style={indexStyles.textGrey}>Sad</Text>
          <FeelingsSlider setFeelingState={setSad} feelingState={sad} />
        </View>

        <View style={[styles.topAndBottom, indexStyles.centering]}>
          <Text style={indexStyles.textGrey}>Friendly</Text>
          <FeelingsSlider
            setFeelingState={setFriendly}
            feelingState={friendly}
          />
        </View>
      </ScrollView>

      {/* Render Continue button  */}
      <TouchableOpacity
        style={[styles.continueButton, indexStyles.centering]}
        onPress={() => {
          handleOnContinuePress();
        }}
      >
        <Text style={styles.continueText}>Continue</Text>
      </TouchableOpacity>
    </View>
  );
}

export default LogFeelingScreen;

const styles = StyleSheet.create({
  headerText: {
    color: "white",
    fontSize: 20,
    fontWeight: "bold",
  },
  header: {
    backgroundColor: colors.darkBorder,
    height: "12%",
    width: "100%",
    borderBottomLeftRadius: 50,
    borderBottomRightRadius: 50,
    alignSelf: "flex-start",
    paddingTop: 30,
  },
  container: {
    alignItems: "center",
    height: "100%",
    justifyContent: "space-between",
    backgroundColor: "#ffd390",
    paddingBottom: "5%",
  },
  continueButton: {
    width: "40%",
    height: "6%",
    backgroundColor: "#c7ffd8",
    borderRadius: 20,
    alignSelf: "center",
  },
  slidersContainer: {
    backgroundColor: "white",
    borderRadius: 50,
    paddingTop: 20,
    borderColor: colors.darkBorder,
    borderWidth: 2,
    width: "100%",
  },
  topAndBottom: {
    marginTop: 10,
  },
  continueText: {
    color: "black",
    fontSize: 20,
    textAlign: "center",
  },
});
