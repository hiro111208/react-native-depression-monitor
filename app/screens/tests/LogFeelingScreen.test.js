import React from "react";
import { shallow } from "enzyme";
import { Alert } from "react-native";
import { render, fireEvent, waitFor } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import LogFeelingScreen from "../LogFeelingScreen";
import * as firebase from "firebase";

describe("Testing LogFeelingScreen.js rendering", () => {

    beforeAll(() => {
        jest.spyOn(firebase, "auth").mockImplementation(() => {
            return {
                currentUser: {
                    uid: "123456",
                },
            }
        })
    });
        

    it("LogFeelingScreen shallow renders correctly", () => {
        const wrapper = shallow(<LogFeelingScreen route={{ params: { userData: { userID: "123456", question: 0}}}}/>);
        expect(wrapper.length).toBe(1);
    });

    it("LogFeelingScreen renders correctly", () => {
        const tree = renderer.create(<LogFeelingScreen route={{ params: { userData: { userID: "123456", question: 0}}}}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("Renders continue button", () => {
        const wrapper = render(<LogFeelingScreen route={{ params: { userData: { userID: "123456", question: 0}}}}/>);
        expect(wrapper.queryByTestId("CONTINUE_BUTTON")).toBeTruthy();
    });

});

//TODO: Add tests for branches
describe("Testing handleOnContinuePress function", () => {

    it("Should alert user", async () => {
        const { getByTestId } = render(<LogFeelingScreen route={{ params: { userData: { userID: "123456", question: 0}}}}/>);
        const button = getByTestId("CONTINUE_BUTTON");

        jest.spyOn(Alert, "alert");
        await waitFor(()=> {
            fireEvent.press(button);
            expect(Alert.alert).toHaveBeenCalledWith("Please fill in how you are feeling!");
        })
    });

});