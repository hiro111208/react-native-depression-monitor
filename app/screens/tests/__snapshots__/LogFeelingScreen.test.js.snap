// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing LogFeelingScreen.js rendering LogFeelingScreen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffd390",
      "height": "100%",
      "justifyContent": "space-between",
      "paddingBottom": "5%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "flex-start",
          "backgroundColor": "#ffa351ff",
          "borderBottomLeftRadius": 50,
          "borderBottomRightRadius": 50,
          "height": "12%",
          "paddingTop": 30,
          "width": "100%",
        },
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      How are you feeling?
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "#ffa351ff",
        "borderRadius": 50,
        "borderWidth": 2,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 10,
        "width": "95%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="VERY_SAD"
    >
      <Image
        source={
          Object {
            "testUri": "../../../app/assets/images/verySad.png",
          }
        }
        style={
          Object {
            "height": 50,
            "marginHorizontal": 7,
            "marginVertical": 7,
            "width": 50,
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="SAD"
    >
      <Image
        source={
          Object {
            "testUri": "../../../app/assets/images/sad.png",
          }
        }
        style={
          Object {
            "height": 50,
            "marginHorizontal": 7,
            "marginVertical": 7,
            "width": 50,
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="NEUTRAL"
    >
      <Image
        source={
          Object {
            "testUri": "../../../app/assets/images/neutral.png",
          }
        }
        style={
          Object {
            "height": 50,
            "marginHorizontal": 7,
            "marginVertical": 7,
            "width": 50,
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="HAPPY"
    >
      <Image
        source={
          Object {
            "testUri": "../../../app/assets/images/happy.png",
          }
        }
        style={
          Object {
            "height": 50,
            "marginHorizontal": 7,
            "marginVertical": 7,
            "width": 50,
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="VERY_HAPPY"
    >
      <Image
        source={
          Object {
            "testUri": "../../../app/assets/images/veryHappy.png",
          }
        }
        style={
          Object {
            "height": 50,
            "marginHorizontal": 7,
            "marginVertical": 7,
            "width": 50,
          }
        }
      />
    </View>
  </View>
  <RCTScrollView
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "#ffa351ff",
        "borderRadius": 50,
        "borderWidth": 2,
        "paddingTop": 20,
        "width": "95%",
      }
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "marginTop": 10,
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "grey",
              "fontSize": 18,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Anxious
        </Text>
        <View
          style={
            Object {
              "width": "85%",
            }
          }
        >
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "marginHorizontal": 16,
                  "marginTop": 12,
                },
                Object {
                  "marginBottom": 44,
                },
              ]
            }
          >
            <RCTSlider
              disabled={false}
              enabled={true}
              maximumTrackTintColor="#ddd"
              maximumValue={5}
              minimumTrackTintColor="#ffd390"
              minimumValue={1}
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onSlidingComplete={[Function]}
              onStartShouldSetResponder={[Function]}
              onValueChange={[Function]}
              step={1}
              style={
                Object {
                  "height": 40,
                }
              }
              testID="SLIDER"
              thumbTintColor="#ffa351ff"
              value={1}
            />
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 44,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontSize": 16,
                      "position": "absolute",
                      "textAlign": "center",
                    },
                    Object {
                      "color": "#ffd390",
                      "transform": Array [
                        Object {
                          "translateX": -0,
                        },
                      ],
                      "width": 0,
                    },
                    Object {
                      "left": 5,
                    },
                  ]
                }
              >
                Not at all
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "fontSize": 16,
                      "position": "absolute",
                      "textAlign": "center",
                    },
                    Object {
                      "color": "#ddd",
                      "transform": Array [
                        Object {
                          "translateX": -0,
                        },
                      ],
                      "width": 0,
                    },
                    Object {
                      "left": 0,
                    },
                  ]
                }
              >
                Somewhat
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "fontSize": 16,
                      "position": "absolute",
                      "textAlign": "center",
                    },
                    Object {
                      "color": "#ddd",
                      "transform": Array [
                        Object {
                          "translateX": -0,
                        },
                      ],
                      "width": 0,
                    },
                    Object {
                      "left": -5,
                    },
                  ]
                }
              >
                Very much
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "marginTop": 10,
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "grey",
              "fontSize": 18,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Sad
        </Text>
        <View
          style={
            Object {
              "width": "85%",
            }
          }
        >
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "marginHorizontal": 16,
                  "marginTop": 12,
                },
                Object {
                  "marginBottom": 44,
                },
              ]
            }
          >
            <RCTSlider
              disabled={false}
              enabled={true}
              maximumTrackTintColor="#ddd"
              maximumValue={5}
              minimumTrackTintColor="#ffd390"
              minimumValue={1}
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onSlidingComplete={[Function]}
              onStartShouldSetResponder={[Function]}
              onValueChange={[Function]}
              step={1}
              style={
                Object {
                  "height": 40,
                }
              }
              testID="SLIDER"
              thumbTintColor="#ffa351ff"
              value={1}
            />
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 44,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontSize": 16,
                      "position": "absolute",
                      "textAlign": "center",
                    },
                    Object {
                      "color": "#ffd390",
                      "transform": Array [
                        Object {
                          "translateX": -0,
                        },
                      ],
                      "width": 0,
                    },
                    Object {
                      "left": 5,
                    },
                  ]
                }
              >
                Not at all
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "fontSize": 16,
                      "position": "absolute",
                      "textAlign": "center",
                    },
                    Object {
                      "color": "#ddd",
                      "transform": Array [
                        Object {
                          "translateX": -0,
                        },
                      ],
                      "width": 0,
                    },
                    Object {
                      "left": 0,
                    },
                  ]
                }
              >
                Somewhat
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "fontSize": 16,
                      "position": "absolute",
                      "textAlign": "center",
                    },
                    Object {
                      "color": "#ddd",
                      "transform": Array [
                        Object {
                          "translateX": -0,
                        },
                      ],
                      "width": 0,
                    },
                    Object {
                      "left": -5,
                    },
                  ]
                }
              >
                Very much
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "marginTop": 10,
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "grey",
              "fontSize": 18,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Happy
        </Text>
        <View
          style={
            Object {
              "width": "85%",
            }
          }
        >
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "marginHorizontal": 16,
                  "marginTop": 12,
                },
                Object {
                  "marginBottom": 44,
                },
              ]
            }
          >
            <RCTSlider
              disabled={false}
              enabled={true}
              maximumTrackTintColor="#ddd"
              maximumValue={5}
              minimumTrackTintColor="#ffd390"
              minimumValue={1}
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onSlidingComplete={[Function]}
              onStartShouldSetResponder={[Function]}
              onValueChange={[Function]}
              step={1}
              style={
                Object {
                  "height": 40,
                }
              }
              testID="SLIDER"
              thumbTintColor="#ffa351ff"
              value={1}
            />
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 44,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontSize": 16,
                      "position": "absolute",
                      "textAlign": "center",
                    },
                    Object {
                      "color": "#ffd390",
                      "transform": Array [
                        Object {
                          "translateX": -0,
                        },
                      ],
                      "width": 0,
                    },
                    Object {
                      "left": 5,
                    },
                  ]
                }
              >
                Not at all
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "fontSize": 16,
                      "position": "absolute",
                      "textAlign": "center",
                    },
                    Object {
                      "color": "#ddd",
                      "transform": Array [
                        Object {
                          "translateX": -0,
                        },
                      ],
                      "width": 0,
                    },
                    Object {
                      "left": 0,
                    },
                  ]
                }
              >
                Somewhat
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "fontSize": 16,
                      "position": "absolute",
                      "textAlign": "center",
                    },
                    Object {
                      "color": "#ddd",
                      "transform": Array [
                        Object {
                          "translateX": -0,
                        },
                      ],
                      "width": 0,
                    },
                    Object {
                      "left": -5,
                    },
                  ]
                }
              >
                Very much
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#c7ffd8",
        "borderRadius": 20,
        "height": "6%",
        "justifyContent": "center",
        "opacity": 1,
        "width": "40%",
      }
    }
    testID="CONTINUE_BUTTON"
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 20,
          "textAlign": "center",
        }
      }
    >
      Continue
    </Text>
  </View>
</View>
`;
